constraint/pkg/client/drivers/k8scel/driver.go

```diff
@@ -33,166 +45,382 @@ package k8scel
+		operationOptions: options,
+	}, nil
+}
+
+func (w *RequestWrapper) GetName() string {
+	return w.ar.Name
+}
+
+func (w *RequestWrapper) GetNamespace() string {
+	return w.ar.Namespace
+}
+
+func (w *RequestWrapper) GetResource() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    w.ar.Resource.Group,
+		Version:  w.ar.Resource.Version,
+		Resource: w.ar.Resource.Resource,
+	}
+}
+
+func (w *RequestWrapper) GetSubresource() string {
+	return w.ar.SubResource
+}
+
+var opMap = map[admissionv1.Operation]admission.Operation{
+	admissionv1.Create:  admission.Create,
+	admissionv1.Update:  admission.Update,
+	admissionv1.Delete:  admission.Delete,
+	admissionv1.Connect: admission.Connect,
+}
+
+func (w *RequestWrapper) GetOperation() admission.Operation {
+	return opMap[w.ar.Operation]
+}
+
+func (w *RequestWrapper) GetOperationOptions() runtime.Object {
+	return w.operationOptions
+}
+
+func (w *RequestWrapper) IsDryRun() bool {
+	if w.ar.DryRun == nil {
+		return false
+	}
+	return *w.ar.DryRun
+}
+
+func (w *RequestWrapper) GetObject() runtime.Object {
+	return w.object
+}
+
+func (w *RequestWrapper) GetOldObject() runtime.Object {
+	return w.oldObject
+}
+
+func (w *RequestWrapper) GetKind() schema.GroupVersionKind {
+	return schema.GroupVersionKind{
+		Group:   w.ar.Kind.Group,
+		Version: w.ar.Kind.Version,
+		Kind:    w.ar.Kind.Kind,
+	}
+}
+
+func (w *RequestWrapper) GetUserInfo() user.Info {
+	extra := map[string][]string{}
+	for k := range w.ar.UserInfo.Extra {
+		vals := make([]string, len(w.ar.UserInfo.Extra[k]))
+		copy(vals, w.ar.UserInfo.Extra[k])
+		extra[k] = vals
+	}
+
+	return &user.DefaultInfo{
+		Name:   w.ar.UserInfo.Username,
+		UID:    w.ar.UserInfo.UID,
+		Groups: w.ar.UserInfo.Groups,
+		Extra:  extra,
+	}
+}
+
+func (w *RequestWrapper) AddAnnotation(key, value string) error {
+	return errors.New("AddAnnotation not implemented")
+}
+
+func (w *RequestWrapper) AddAnnotationWithLevel(key, value string, level auditinternal.Level) error {
+	return errors.New("AddAnnotationWithLevel not implemented")
+}
+
+func (w *RequestWrapper) GetReinvocationContext() admission.ReinvocationContext {
+	return nil
+}
+
+func convertv1alpha1FailurePolicyTypeTov1FailurePolicyType(policyType *admissionv1alpha1.FailurePolicyType) *admissionregistrationv1.FailurePolicyType {
+	if policyType == nil {
+		return nil
+	}
+
+	var v1FailPolicy admissionregistrationv1.FailurePolicyType
+	if *policyType == admissionv1alpha1.Fail {
+		v1FailPolicy = admissionregistrationv1.Fail
+	} else if *policyType == admissionv1alpha1.Ignore {
+		v1FailPolicy = admissionregistrationv1.Ignore
+	}
+	return &v1FailPolicy
+}
+
+func convertv1alpha1Validations(inputValidations []admissionv1alpha1.Validation) []cel.ExpressionAccessor {
+	celExpressionAccessor := make([]cel.ExpressionAccessor, len(inputValidations))
+	for i, validation := range inputValidations {
+		validation := validatingadmissionpolicy.ValidationCondition{
```

Comment:
nit: to enhance readability, can we use a different var name for the inner validation variable?
```suggestion
		celValidation := validatingadmissionpolicy.ValidationCondition{
```